Sources/SpeziFHIR/FHIRStore.swift

```diff
@@ -19,149 +17,146 @@ import Spezi
 /// The ``FHIRStore`` is automatically injected in the environment if you use the ``FHIR`` standard or can be used as a standalone module.
 @Observable
 public final class FHIRStore: Module,
-                              EnvironmentAccessible,
-                              DefaultInitializable,
-                              @unchecked Sendable /* `unchecked` `Sendable` conformance fine as access to `_resources` protected by `NSLock` */ {
-    private let lock = NSLock()
-    @ObservationIgnored private var _resources: [FHIRResource]
-    
-    
-    /// Allergy intolerances.
+                             EnvironmentAccessible,
+                             DefaultInitializable,
+                             Sendable {
+    /// Actor-isolated storage for `FHIRResource`s backing the ``FHIRStore``.
+    private actor Storage {
+        // Non-isolation required so that resource access via the `FHIRStore` stays sync (required for seamless SwiftUI access).
+        // Isolation is still guaranteed as the only modifying functions `insert()` and `remove()` are isolated on the `FHIRStore.Storage` actor.
+        nonisolated(unsafe) private var _resources: [FHIRResource] = []
```

Comment:
Maybe we can use a dict for O(1) lookup instead of an array? WDYT?

Reply:
`FHIRResource`s are basically never accessed via their identifier, at least in the use cases that we currently have. Therefore, I'm totally fine with an array here.

Reply:
Operations in `remove(resource:)`, `fetch(for:)` and in the subscript are O(n).

Reply:
Yes, of course, individual access is O(n) in an array. However, as mentioned, individual FHIR resources are basically never accessed by their identifier (not that I've seen at least for now), making the array a valid choice here (the current implementation actually is an array, so I don't see a good enough reason that change it)

If we would store the `FHIRResource`s as a Dictionary, all subset operations (which are the actually heavily needed ones, see the individual categories) would first need to get the array representation of the dictionary (overhead). Also, it might affect the order of the array-based representation, as soon as the dictionary is modified. This might break some use cases of apps relying on a stable ordering, even after insertion.